crates/aptos/src/node/mod.rs

```diff
@@ -231,6 +235,59 @@ impl ValidatorNetworkAddressesArgs {
     }
 }
 
+#[derive(Parser)]
+pub struct GetPoolAddress {
+    #[clap(long)]
+    pub(crate) owner_address: AccountAddress,
+    #[clap(long)]
+    pub(crate) operator_address: Option<AccountAddress>,
+    #[clap(long)]
+    pub(crate) employee_account_index: Option<u64>,
+    #[clap(flatten)]
+    pub(crate) rest_options: RestOptions,
+    #[clap(flatten)]
+    pub(crate) profile_options: ProfileOptions,
+}
+
+#[async_trait]
+impl CliCommand<AccountAddress> for GetPoolAddress {
+    fn command_name(&self) -> &'static str {
+        "GetPoolAddress"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<AccountAddress> {
+        let owner_address = self.owner_address;
+
+        let client = self.rest_options.client(&self.profile_options.profile)?;
+        let stake_pool_exists = client
+            .get_account_resource_bcs::<StakePool>(owner_address, "0x1::stake::StakePool")
+            .await
+            .is_ok();
+        let stake_pool_address = if stake_pool_exists {
+            owner_address
+        } else {
+            let staking_contract_store_exists = client
+                .get_resource::<Vec<u8>>(owner_address, "0x1::staking_contract::Store")
+                .await
+                .is_ok();
+            let operator_address = self
+                .operator_address
+                .unwrap_or(self.profile_options.account_address()?);
+            if staking_contract_store_exists {
+                default_stake_pool_address(owner_address, operator_address)
+            } else {
+                create_vesting_pool_address(
+                    owner_address,
+                    operator_address,
+                    self.employee_account_index.unwrap(),
+                    &[],
+                )
+            }
```

Comment:
Maybe we should just show all the staking pools if there is no operator address provided.

Reply:
That'd take more work as we'd need to parse the SimpleMap. I can do that in a separate PR later